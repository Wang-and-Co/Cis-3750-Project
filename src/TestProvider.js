/* eslint-disable react/destructuring-assignment */
import * as React from 'react';
import { createTheme } from '@mui/material';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
// import defaultTheme from '../../src/renderer/data/theme';

export default class TestProvider extends React.Component {
  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children,
    },
  };

  render() {
    const theme = createTheme({
      //   ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions,
      },
    });
    return (
      // eslint-disable-next-line react/jsx-no-comment-textnodes
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
      </StyledEngineProvider>
    );
  }
}
