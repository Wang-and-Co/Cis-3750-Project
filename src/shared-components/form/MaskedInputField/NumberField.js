import { useField } from 'formik';
import InputField from '../InputField/InputField';

const NumberField = ({ name, ...otherProps }) => {
  const [field] = useField({ name: name });

  const sanitizeInput = (event) => {
    const newValue = event.target.value;
    //
    const matches = newValue.match(/-{0,1}[0-9]*\.{0,1}[0-9]*/);
    if (matches?.[0]?.localeCompare(newValue) === 0) {
      field.onChange(event);
    }
  };
  return (
    <InputField
      {...otherProps}
      name={name}
      onChange={sanitizeInput}
    ></InputField>
  );
};
export default NumberField;
