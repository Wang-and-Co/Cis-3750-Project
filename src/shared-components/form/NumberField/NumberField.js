import { useField } from 'formik';
import InputField from '../InputField/InputField';
import PropTypes from 'prop-types';

const NumberField = ({ name, maxLength = 120, ...otherProps }) => {
  const [field] = useField({ name: name });

  // intercept the onChange, handle updating the text here
  const sanitizeInput = (event) => {
    const newValue = event.target.value;
    const matches = newValue.match(/-{0,1}[0-9]*\.{0,1}[0-9]*/); //format of

    if (matches?.[0]?.localeCompare(newValue) !== 0) return;

    const numberSections = newValue.match(/(-{0,1})([0-9]*)\.([0-9]*)/);
    const finalValue =
      numberSections !== null && numberSections[2] === ''
        ? numberSections[1] + '0.' + numberSections[3]
        : newValue;

    if (finalValue.length >= maxLength) return;

    event.target.value = finalValue;
    field.onChange(event);
  };
  return (
    <InputField
      {...otherProps}
      name={name}
      onChange={sanitizeInput}
    ></InputField>
  );
};

NumberField.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  helperText: PropTypes.string,
  required: PropTypes.bool,
  maxLength: PropTypes.number,
};
export default NumberField;
