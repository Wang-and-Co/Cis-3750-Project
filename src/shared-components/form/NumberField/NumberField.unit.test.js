import { create } from 'react-test-renderer';
import TestProvider from '../../../TestProvider';
import { Form, Formik } from 'formik';
import NumberField from './NumberField';
const props = {
  name: 'test',
  label: 'Hello there',
  required: false,
  helperText: 'among us',
  variant: 'filled',
};

let inputFieldRenderer;
let inputFieldInstance;

beforeEach(async () => {
  await renderGenericModal(props);
});

afterEach(() => {
  inputFieldRenderer.unmount();
});
describe('NumberField', () => {
  it('renders correctly', () => {
    expect(inputFieldRenderer.toJSON()).toMatchSnapshot();
  });
  it('passes the correct props', () => {
    expect(GetInputProp('name')).toBe(props.name);
    expect(GetInputProp('label')).toBe(props.label);
    expect(GetInputProp('required')).toBe(props.required);
    expect(GetInputProp('helperText')).toBe(props.helperText);
    expect(GetInputProp('variant')).toBe(props.variant);
  });
});

const GetInputProp = (prop) =>
  inputFieldInstance.findByType(NumberField).props[prop];

const renderGenericModal = async (editorProps) => {
  inputFieldRenderer = await create(
    <TestProvider>
      <Formik>
        <Form>
          <NumberField {...editorProps}></NumberField>
        </Form>
      </Formik>
    </TestProvider>,
  );
  inputFieldInstance = inputFieldRenderer.root;
};
