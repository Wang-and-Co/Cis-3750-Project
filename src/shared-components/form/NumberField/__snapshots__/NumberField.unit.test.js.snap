// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberField renders correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u6ro6r-MuiFormControl-root-MuiTextField-root"
    onKeyDown={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="input-test"
      id="input-test-label"
    >
      Hello there
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-batk84-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-describedby="input-test-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id="input-test"
        name="test"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      id="input-test-helper-text"
    >
      among us
    </p>
  </div>
</form>
`;
